### ISSUE TITLE
[Bug]: Static Page Generator with sites are not dynamic with Nginx setup

### ISSUE THREAD

**Issue by Gerhard-Kanzler** [pimcore/pimcore#15511]
### Pimcore version

10.x/11.x

### Steps to reproduce

When in Pimcore a site is defined the static page is in the folder called the site
/pages/my-site/en/portal

But the domain in the site config is maybe "pimcore.com"
/pages/pimcore.com/en/portal

The better solution is a option to use the domain or a custom string (maybe switch) for generating the folder structure in pages folder.

For dynamic use in Nginx the mapping on domain names are a better solution.

**Comment by brusch** [comment]
I think a config on site level would be a good solution, where you can either use the main domain of the site or a custom flysystem service id or the default as it is now.

**Comment by dvesh3** [comment]
Current workaround to this issue is to use enable `static_page_router`:
```
pimcore:
    documents:
        static_page_router:
            enabled: true
```
and the page is resolved based on site context as fixed [here](https://github.com/pimcore/pimcore/pull/11810).

@brusch should we try to solve this with server returning the pages directly as requested in this issue or adding some docs is enough?

**Comment by dvesh3** [comment]
Fixed by #16030
